code/modules/mob/new_player/new_player.dm

```diff
@@ -329,12 +329,19 @@
 	qdel(src)
 
 /mob/new_player/proc/AnnounceArrival(var/mob/living/carbon/human/character, var/rank)
-	if (ticker.current_state == GAME_STATE_PLAYING)
-		if(announcement_systems.len)
-			if(character.mind)
-				if((character.mind.assigned_role != "Cyborg") && (character.mind.assigned_role != character.mind.special_role))
-					var/obj/machinery/announcement_system/announcer = pick(announcement_systems)
-					announcer.announce("ARRIVAL", character.real_name, rank, list()) //make the list empty to make it announce it in common
+	if(ticker.current_state != GAME_STATE_PLAYING)
+		return
+	var/message = "<span class='game deadsay'><span class='name'>\
+		[character.real_name]</span> ([rank]) has arrived at the station.\
+		</span>"
+	deadchat_broadcast(message, follow_target = character)
+	if((!announcement_systems.len) || (!character.mind))

```

Comment:
You're gonna want to have the !character.mind check BEFORE the deadchat broadcast, unless you want deadchat to be spammed whenever an event spawns an NPC.